src/components/input/code/views/inputView.jsx

```diff
@@ -0,0 +1,82 @@
+var React = require('react');
+
+module.exports = React.createClass({
+
+	/**
+	 * @function focus
+	 * @description Focus on the element.
+	 */
+	focus: function() {
+		if (this.isMounted()) {
+			this.refs.input.getDOMNode().focus();
+		}
+	},
+
+	/**
+	 * @function value
+	 * @returns {string} The value of the input.
+	 */
+	value: function() {
+		return this.state.value;
+	},
+
+	/**
+	 * @function clear
+	 * @description Clear the value of the element.
+	 */
+	clear: function() {
+		this.setState({
+			value: ''
+		});
+	},
+
+	/**
+	 * @private
+	 * @function onChange
+	 * @descriptin When the value of the input changes, update the state.
+	 */
+	_onChange: function(event) {
+
+		// Get the value from the input
+		var value = event.target.value;
+
+		// Set the state
+		this.setState({
+			value: value
+		});
+
+		// Run the onChange function if it exists
+		if (this.props.onChange) {
+			this.props.onChange(event, value);
+		}
+
+	},
+
+	propTypes: {
+		label: React.PropTypes.string,
+		type: React.PropTypes.string,
```

Comment:
Currently I can pass any input type I want, might be worth using `React.PropTypes.oneOf([ ... ])` to lock it down to the types this component supports.